src/store/modules/snackbar.js

```diff
@@ -14,20 +22,42 @@ const mutations = {
     state.visible = false
     state.message = null
   },
+  setProcessing(state, status) {
+    state.processing = status
+  },
 }
 
 const actions = {
-  async show({ state, commit }, { message }) {
-    if (state.visible) {
-      commit('hide')
-      await new Promise((r) => setTimeout(r, 200))
+  async show({ state, commit, dispatch }, { message }) {
+    commit('enqueue', message)
+    if (!state.processing) {
+      dispatch('processQueue')
     }
-    commit('show', message)
   },
-  hide({ state, commit }) {
-    if (state.visible) {
-      commit('hide')
+
+  async processQueue({ state, commit, dispatch }) {
```

Comment:
What is the reason for introducing the `processQueue` action?
Is it intended to be called by a user? If not, in my opinion, it would be cleaner to integrate its functionality into the already existing `show` action.